networkx/algorithms/shortest_paths/generic.py

```diff
@@ -585,11 +585,8 @@ def single_source_all_shortest_paths(G, source, weight=None, method="dijkstra"):
         pred, dist = nx.bellman_ford_predecessor_and_distance(G, source, weight=weight)
     else:
         raise ValueError(f"method not supported: {method}")
-    for n in G:
-        try:
-            yield n, list(_build_paths_from_predecessors({source}, n, pred))
-        except nx.NetworkXNoPath:
-            pass
+    for n in pred:
```

Comment:
In my AoC code, your proposed change is twice slower than iterating just on `pred`. (But it's still 13 times faster than the current code.) But what bothers me most is that it would make the performance of this function still depend on the total number of nodes in the graph instead of just the number of reachable nodes, which can be orders of magnitude smaller, or even change the complexity of the function.

Is it really that important to keep the dict order the same as the whole graph? I mean, I've grepped the doc for mentions of node ordering, I see some mentions of insertion order being kept for the whole graph, but not for functions results. For instance, the various predecessor functions build their dict in order of exploration.

If it's ok, I'd rather change the docstring to fix the example and maybe state that there is no particular order or that the order has changed in *next version number*.

However, if the node order is very important for some reason, then maybe it should be handled explicitly? Like having an attribute that records the insertion order of nodes and edges. But IMO, this would be cumbersome for very little gain.

Reply:
Yes -- you are right.  There is no implied contract for the order of the returned values .  Indeed, as the failing example shows, it actually makes more sense to have the nodes appear in BFS order rather than in the order that the original graph would report them.  Thanks for pushing back on my comments here!! I just saw the failures and didn't look at the big picture.

Can you please change line 375 of that file to switch the dict entries for '2' and '3'?
I can do it if you prefer. You've done a lot already. :) But let me know. Once we get the green checkmarks this should be ready to for approval.

Reply:
I guess you meant line 562? xD
I rebased on the new main and force-pushed my branch. It's all green now.